test/node-rpc-test.js

```diff
@@ -245,6 +262,41 @@ describe('RPC', function() {
       assert.equal(node.chain.tip.version, info.version);
     });
 
+    it('should rpc getblockbyheight', async () => {
+      // Getting block height of chain tip
+      const blockheight = node.chain.tip.height;
+      const blockHash = node.chain.tip.hash;
+
+      // verbose=0 details=0
+      let blockInfo;
+      blockInfo = await nclient.execute('getblockbyheight', [blockheight, 0, 0]);
+      const block = Block.fromRaw(Buffer.from(blockInfo, 'hex'));
+      assert.bufferEqual(blockHash, block.hash());
+
+      // verbose=1 details=0
+      blockInfo = await nclient.execute('getblockbyheight', [blockheight, 1, 0]);
+      const properties = [
+        'hash', 'confirmations', 'strippedsize',
+        'size', 'weight', 'height', 'version',
+        'versionHex', 'merkleroot', 'coinbase',
+        'tx', 'time', 'mediantime', 'nonce',
+        'bits', 'difficulty', 'chainwork',
+        'nTx', 'previousblockhash', 'nextblockhash'
+      ];
+      for (const property of properties)
+        assert(property in blockInfo);
+
+      assert.bufferEqual(blockHash, block.hash());
+
+      // verbose=1 details=1
+      blockInfo = await nclient.execute('getblockbyheight', [blockheight, 1, 1]);
+      for (const property of properties)
+        assert(property in blockInfo);
+
+      assert.strictEqual(typeof blockInfo.tx[0], 'object');
```

Comment:
Maybe this can be 

```js
assert(Array.isArray(blockInfo.tx))
```

and for the previous test, check using `Number.isInteger()`

Reply:
I didn't understand that we were getting an array of tx objects in this case and in the previous test, we were getting an array of txs. So how can we check it using ```assert(Array.isArray(blockInfo.tx))```  as it will give the same result in both cases and where we can use ```Number.isInteger()```

Reply:
oh yes you are absolutely right, apologies! Maybe what we should do is add

```js
assert.strictEqual(typeof blockInfo.tx[0], 'string');
```

On line 290 or so.